File: server/internal/models/note.go
package models

import "time"

type Note struct {
	ID        string    `json:"id"`
	Title     string    `json:"title"`
	Content   string    `json:"content"`
	Tags      []string  `json:"tags"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type CreateNoteInput struct {
	Title   string   `json:"title" binding:"required,min=1"`
	Content string   `json:"content"`
	Tags    []string `json:"tags"`
}

type UpdateNoteInput struct {
	Title   string   `json:"title" binding:"required,min=1"`
	Content string   `json:"content"`
	Tags    []string `json:"tags"`
}

File: frontend/src/types/note.ts
export interface Note {
  id: string
  title: string
  content: string
  tags: string[]
  created_at: string
  updated_at: string
}

export interface CreateNoteData {
  title: string
  content: string
  tags: string[]
}

export interface UpdateNoteData {
  title: string
  content: string
  tags: string[]
}
